public/app/plugins/datasource/expr/ExpressionDatasource.ts

```diff
@@ -0,0 +1,77 @@
+import { DataSourceApi, DataQueryRequest, DataQueryResponse, DataSourceInstanceSettings } from '@grafana/ui';
+import { ExpressionQuery, GELQueryType } from './types';
+import { ExpressionQueryEditor } from './ExpressionQueryEditor';
+import { Observable, from } from 'rxjs';
+import { getBackendSrv, config } from '@grafana/runtime';
+import { gelResponseToDataFrames } from './util';
+
+/**
+ * This is a singleton that is not actually instanciated
+ */
+export class ExpressionDatasourceApi extends DataSourceApi<ExpressionQuery> {
+  constructor(instanceSettings: DataSourceInstanceSettings) {
+    super(instanceSettings);
+  }
+
+  getCollapsedText(query: ExpressionQuery) {
+    return `Expression: ${query.type}`;
+  }
+
+  query(request: DataQueryRequest): Observable<DataQueryResponse> {
+    const { targets, intervalMs, maxDataPoints, range } = request;
+
+    const orgId = (window as any).grafanaBootData.user.orgId;
+    const queries = targets.map(q => {
+      if (q.datasource === ExpressionDatasourceID) {
+        return {
+          ...q,
+          datasourceId: this.id,
+          orgId,
+        };
+      }
+      const ds = config.datasources[q.datasource || config.defaultDatasource];
+      return {
+        ...q,
+        datasourceId: ds.id,
+        intervalMs,
+        maxDataPoints,
+        orgId,
+        // ?? alias: templateSrv.replace(q.alias || ''),
+      };
+    });
+
+    return from(
+      getBackendSrv()
+        .post('/api/ds', {
```

Comment:
@kylebrandt @marefr -- note this uses `/api/ds` we should change that to whatever matches the new endpoint

Reply:
We can use whatever we find suitable. The one that exists now is a v2/copy of tsdb query and we probably want to use something else :+1:

Reply:
should this not use datasourceRequest() function (otherwise query inspector does not work)